MicrosoftDX/nether
# Analytics as a library

### Issue: #461 


### Description:
Nether Analytics is completely re-written and cleaned up. The new version can now be consumed both as a library and as an "out of the box" experience. Currently Nether.sln has been stripped of all references to Nether.Analytics until we come up with a new solution on how to integrate (Basically there need to be some interfaces that needs to be re-written).

### Test:

Create the following resources in Azure:

* Azure AD Application
* Azure Data Lake Analytics Account
* Azure Data Lake Store Account
* EventHubs Account and a EventHub

This PowerShell Script helps you setup all required environment parameters you need unless you rather implement them in the json-config file.

The script has been modified and doesn't contain correct information to actual resources in Azure.

```
[Environment]::SetEnvironmentVariable("NAH_EHLISTENER_CONNECTIONSTRING", "Endpoint=sb://nether.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=h16jv6nc0bxxxxg0YI14=", "User")
[Environment]::SetEnvironmentVariable("NAH_EHLISTENER_EVENTHUBPATH", "ingest", "User")
[Environment]::SetEnvironmentVariable("NAH_EHLISTENER_CONSUMERGROUP", "nether", "User")
[Environment]::SetEnvironmentVariable("NAH_EHLISTENER_STORAGECONNECTIONSTRING", "DefaultEndpointsProtocol=https;AccountName=netherdashboard;AccountKey=oT30a8/BSwTFg/4GGWLPCeGIHBxxxjtwU9kg9YaL0Qw==;EndpointSuffix=core.windows.net", "User")
[Environment]::SetEnvironmentVariable("NAH_EHLISTENER_LEASECONTAINERNAME", "sync", "User")

[Environment]::SetEnvironmentVariable("NAH_AAD_DOMAIN", "microsoft.onmicrosoft.com", "User")
[Environment]::SetEnvironmentVariable("NAH_AAD_CLIENTID", "b43e3e6c-xxxbd63-a3683b27e546", "User")
[Environment]::SetEnvironmentVariable("NAH_AAD_CLIENTSECRET", "QxoNhF1csZVsfpfPjxxxe0Y3yil9Etnp4bw=", "User")

[Environment]::SetEnvironmentVariable("NAH_AZURE_SUBSCRIPTIONID", "3a856395-xxxxx-527c857e5995", "User")

[Environment]::SetEnvironmentVariable("NAH_AZURE_DLSOUTPUTMANAGER_ACCOUNTNAME", "nether", "User")
```

Once setup:

1. Open and compile the two Solutions that are now available: Nether.sln and Nether.Analytics.sln. They should both compile and work.
2. Run Nether.Analytics.sln with two startup projects: NehterAnalyticsTestClient and Nether.Analytics.Host
3. Start sending geo-location messages (currently only message type implemented) from the test client.
4. Make sure information ends up in your Data Lake Store Account.




(Make sure that all above information is provided before submitting your pull request)
